File: llm.py
"""
Groq LLM client for Pedro Tech Assistant.
"""
import os
from groq import Groq
from dotenv import load_dotenv

load_dotenv()

API_KEY = os.getenv("GROQ_API_KEY") or os.getenv("LLM_API_KEY")
if not API_KEY:
    raise RuntimeError(
        "Missing GROQ_API_KEY. Add it to .env (see .env.example)."
    )

client = Groq(api_key=API_KEY)
MODEL = os.getenv("LLM_MODEL", "qwen/qwen3.6-27b")
TEMPERATURE = float(os.getenv("LLM_TEMPERATURE", "0.6"))
MAX_TOKENS = int(os.getenv("LLM_MAX_TOKENS", "2048"))


def ask_llm(system: str, user: str, label: str = "llm") -> str:
    """Send a system + user message to Groq and return the text reply."""
    print(f"[{label}] → {MODEL} ...", flush=True)
    completion = client.chat.completions.create(
        model=MODEL,
        messages=[
            {"role": "system", "content": system},
            {"role": "user", "content": user},
        ],
        temperature=TEMPERATURE,
        max_completion_tokens=MAX_TOKENS,
        top_p=0.95,
    )
    text = (completion.choices[0].message.content or "").strip()
    print(f"[{label}] ✓", flush=True)
    return text or "No response generated."


def get_config() -> dict:
    return {
        "model": MODEL,
        "temperature": TEMPERATURE,
        "max_tokens": MAX_TOKENS,
    }

File: src/llm.ts
import OpenAI from 'openai';
import dotenv from 'dotenv';
dotenv.config();

const client = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY || process.env.LLM_API_KEY || 'ollama',
  baseURL: process.env.LLM_BASE_URL || 'http://127.0.0.1:11434/v1',
});

/**
 * Thin LLM wrapper (OpenAI-compatible).
 * Works with Ollama, Groq, OpenAI, and other compatible providers.
 */
export async function askLLM(system: string, user: string): Promise<string> {
  const res = await client.chat.completions.create({
    model: process.env.LLM_MODEL || 'llama3.1:8b',
    messages: [
      { role: 'system', content: system },
      { role: 'user', content: user },
    ],
    temperature: 0.3,
  });
  return res.choices[0]?.message?.content?.trim() || 'No response generated.';
}
